dataplane/windows/policysets/static_rules.go

```diff
@@ -0,0 +1,159 @@
+		return nil, fmt.Errorf("'Priority' should not be zero")
+	}
+	if p.Type != hns.ACL {
+		return nil, fmt.Errorf("'Type' is not ACL")
+	}
+	if (p.RuleType != hns.Host) && (p.RuleType != hns.Switch) {
+		return nil, fmt.Errorf("'RuleType' %s is invalid", p.RuleType)
+	}
+	if (p.Action != hns.Allow) && (p.Action != hns.Block) {
+		return nil, fmt.Errorf("'Action' %s is invalid", p.Action)
+	}
+	if (p.Direction != hns.In) && (p.Direction != hns.Out) {
+		return nil, fmt.Errorf("'Direction' %s is invalid", p.Direction)
+	}
+
+	return &hns.ACLPolicy{
+		Type:            p.Type,
+		Id:              prefix + "-" + p.Id,
+		Protocol:        p.Protocol,
+		Action:          p.Action,
+		Direction:       p.Direction,
+		LocalAddresses:  p.LocalAddresses,
+		RemoteAddresses: p.RemoteAddresses,
+		LocalPorts:      p.LocalPorts,
+		RemotePorts:     p.RemotePorts,
+		RuleType:        p.RuleType,
+		Priority:        p.Priority,
+	}, nil
+}
+
+type staticEndpointPolicy struct {
+	Name string        `json:"Name"`
+	Rule staticACLRule `json:"Rule"`
+}
+
+type staticEndpointPolicies struct {
+	Provider string                 `json:"Provider"`
+	Version  string                 `json:"Version"`
+	Rules    []staticEndpointPolicy `json:"Rules"`
+}
+
+// staticRulesReader is a wrapper to read a file.
+// So we can have a mock reader for UT.
+type StaticRulesReader interface {
+	ReadData() ([]byte, error)
+}
+
+type FileReader string
+
+func (f FileReader) ReadData() ([]byte, error) {
+	// The value of os.Args[0] is "c:\CalicoWindows\calico-node.exe" which
+	// is the executable for CalicoFelix service. The static rules file is located
+	// at same directory with "calico-node.exe".
+	rootDir := filepath.Dir(os.Args[0])
+	ruleFile := filepath.Join(rootDir, string(f))
+
+	if _, err := os.Stat(ruleFile); os.IsNotExist(err) {
+		return []byte{}, ErrNoRuleSpecified
+	}
+	return ioutil.ReadFile(ruleFile)
+}
+
+// Read ACL policy rules from static rule file.
+func readStaticRules(r StaticRulesReader) (policies []*hns.ACLPolicy) {
+	data, err := r.ReadData()
+	if err == ErrNoRuleSpecified {
+		log.Info("Ignoring absent static rules file")
+		return
+	}
+	if err != nil {
+		log.WithError(err).Errorf("Failed to read static rules file.")
+		return
+	}
+
+	staticPolicies := staticEndpointPolicies{}
+
+	if err = json.Unmarshal(data, &staticPolicies); err != nil {
+		log.WithError(err).Errorf("Failed to unmarshal static rules file <provider %s, version %s>.",
+			staticPolicies.Provider, staticPolicies.Version)
+		return
+	}
+
+	if len(staticPolicies.Provider) == 0 {
+		log.WithError(err).Errorf("Provider is not specified")
+		return
+	}
+
+	for _, r := range staticPolicies.Rules {
+		if r.Rule.Type != hns.ACL {
+			log.WithField("static rules", r.Rule).Errorf("Incorrect static rule")
+			continue
+		}
+		hnsRule, err := r.Rule.ToHnsACLPolicy(staticPolicies.Provider)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to convert static rule to ACL rule.")
+			continue
```

Comment:
I wonder if we should just quit if we can't parse and apply all of the rules in the static file.